src/core/toast.ts

```diff
@@ -123,7 +123,7 @@ function handlePromise<TData = unknown, TError = unknown, TPending = unknown>(
     // Remove the toast if the input has not been provided. This prevents the toast from hanging
     // in the pending state if a success/error toast has not been provided.
     if (input == null) {
```

Comment:
Instead of writing double if, you can write in this way with single if - 
```js
if (!input && id) {
  toast.dismiss(id);
  return;
}
```
or
```js
if (input == null && id) {
  toast.dismiss(id);
  return;
}
```